viaf.py

```diff
@@ -2,60 +2,271 @@
+        # around the primary picked on, with relevancy weights given by this.
+        return match_confidences["total"]
 
-        if not display_name or not family_name:
-            default_family, default_display = contributor.default_names(None)
-            if not display_name:
-                display_name = default_display
-            if not family_name:
-                family_name = default_family
 
-        self.log.info("VIAF ID: %s" % viaf)
-        if wikipedia_name:
-            self.log.info("Wikipedia name: %s" % wikipedia_name)
-        self.log.info("Display name: %s" % display_name)
-        self.log.info("Family name: %s" % family_name)
-        return viaf, display_name, family_name, wikipedia_name
+    @classmethod
+    def weigh_titles(cls, known_titles=None, contributor_titles=None, match_confidences=None, strict=False):
+        if known_titles:
+            for known_title in known_titles:
+                if strict: 
+                    if known_title in contributor_titles:
+                        match_confidences["title"] = 100
+                        match_confidences["total"] += 0.8 * match_confidences["title"]
+                        # once we find one matching title, no need to keep looking
+                        break
+                else:
+                    for contributor_title in contributor_titles:
+                        # when the second half of the title has something like:
+                        # "Edited by", a colon or semicolon, a bracket or parentheses, a hyphen, 
+                        # one of the institutional authors, like Disney Book Group, elibrary, Inc, 
+                        # Harvard University, Harper & Brothers, 
+                        # then see if can get an exact substring match on the title.
+                        # We want to accept "Pride and Prejudice (Unabridged)" as equivalent to 
+                        # "Pride and Prejudice", but reject "Pride and Prejudice and Zombies" 
+                        # as probably not written by Jane Austen. 
+                        # TODO: In future, consider doing:
+                        # "Pride and Prejudice (Spanish)" should connect to two authors -- 
+                        # Jane Austen and the translator.
+                        if cls.name_matches(unfluff_title(contributor_title), unfluff_title(known_title)):
+                            match_confidences["title"] = 90
+                            match_confidences["total"] += 0.8 * match_confidences["title"]
+                            # match is good enough, we can stop
+                            break
+
+                        '''
+                        Fixes issue where 
+                        <ns1:title>Britain, detente and changing east-west relations</ns1:title> (with accented e in detente)
+                        doesn't match "Britain, Detente and Changing East-West Relations" in our DB.
+                        '''
```

Comment:
And, importantly, if there are _no_ ASCII characters in the title (because it's in a different script) the whole idea of trying to get an ASCII version is junked.